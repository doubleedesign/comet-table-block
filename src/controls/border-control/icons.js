/**
 * WordPress dependencies
 */
import { Path, SVG } from '@wordpress/components';

export const borderSolid = (
	<SVG viewBox="0 0 24 24">
		<Path d="M2,10h20v4H2V10z" />
	</SVG>
);

export const borderDotted = (
	<SVG viewBox="0 0 24 24">
		<Path d="M6.5,12c0-1.4-1.1-2.5-2.5-2.5S1.5,10.6,1.5,12s1.1,2.5,2.5,2.5S6.5,13.4,6.5,12z" />
		<Path d="M14.5,12c0-1.4-1.1-2.5-2.5-2.5S9.5,10.6,9.5,12s1.1,2.5,2.5,2.5S14.5,13.4,14.5,12z" />
		<Path d="M22.5,12c0-1.4-1.1-2.5-2.5-2.5s-2.5,1.1-2.5,2.5s1.1,2.5,2.5,2.5S22.5,13.4,22.5,12z" />
	</SVG>
);

export const borderDashed = (
	<SVG viewBox="0 0 24 24">
		<Path d="M1,10h6v4H1V10z" />
		<Path d="M9,10h6v4H9V10z" />
		<Path d="M17,10h6v4h-6V10z" />
	</SVG>
);

export const borderDouble = (
	<SVG viewBox="0 0 24 24">
		<Path d="M2,9h20v2H2V9z" />
		<Path d="M2,13h20v2H2V13z" />
	</SVG>
);
