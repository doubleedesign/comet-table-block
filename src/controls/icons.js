/**
 * Internal dependencies
 */
import { ViewBox, TopStroke, RightStroke, BottomStroke, LeftStroke } from './styles';

export function SideControlIcon( { sides } ) {
	const top = ! sides || sides.includes( 'top' );
	const right = ! sides || sides.includes( 'right' );
	const bottom = ! sides || sides.includes( 'bottom' );
	const left = ! sides || sides.includes( 'left' );

	return (
		<ViewBox>
			<TopStroke isFocused={ top } />
			<RightStroke isFocused={ right } />
			<BottomStroke isFocused={ bottom } />
			<LeftStroke isFocused={ left } />
		</ViewBox>
	);
}
