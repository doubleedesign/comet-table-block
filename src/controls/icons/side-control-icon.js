/**
 * External dependencies
 */
import { css } from '@emotion/react';
import styled from '@emotion/styled';

/**
 * Internal dependencies
 */
import { ViewBox } from '../styles/common-styles';

const strokeFocus = ( { isFocused = false } ) => {
	return css( {
		backgroundColor: 'currentColor',
		opacity: isFocused ? 1 : 0.3,
	} );
};

const Stroke = styled.span`
	box-sizing: border-box;
	display: block;
	pointer-events: none;
	position: absolute;
	${ strokeFocus };
`;

const TopStroke = styled( Stroke )`
	height: 2px;
	left: 6px;
	right: 6px;
	top: 3px;
`;

const RightStroke = styled( Stroke )`
	bottom: 6px;
	top: 6px;
	width: 2px;
	right: 3px;
`;

const BottomStroke = styled( Stroke )`
	height: 2px;
	left: 6px;
	right: 6px;
	bottom: 3px;
`;

const LeftStroke = styled( Stroke )`
	bottom: 6px;
	top: 6px;
	width: 2px;
	left: 3px;
`;

export default function SideControlIcon( { sides } ) {
	const top = ! sides || sides.includes( 'top' );
	const right = ! sides || sides.includes( 'right' );
	const bottom = ! sides || sides.includes( 'bottom' );
	const left = ! sides || sides.includes( 'left' );

	return (
		<ViewBox>
			<TopStroke isFocused={ top } />
			<RightStroke isFocused={ right } />
			<BottomStroke isFocused={ bottom } />
			<LeftStroke isFocused={ left } />
		</ViewBox>
	);
}
