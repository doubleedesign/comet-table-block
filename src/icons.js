import { Path, SVG, Polygon } from '@wordpress/components';

export const alignTop = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="M9 20h6V9H9v11zM4 4v1.5h16V4H4z" />
	</SVG>
);

export const alignBottom = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="M15 4H9v11h6V4zM4 18.5V20h16v-1.5H4z" />
	</SVG>
);

export const alignMiddle = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="M20 11h-5V4H9v7H4v1.5h5V20h6v-7.5h5z" />
	</SVG>
);

export const splitCells = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="m22.24979,5.97071a1.20586,1.20586 0 0 0 -1.20586,-1.20586l-18.08787,0a1.20586,1.20586 0 0 0 -1.20586,1.20586l0,12.05858a1.20586,1.20586 0 0 0 1.20586,1.20586l18.08787,0a1.20586,1.20586 0 0 0 1.20586,-1.20586l0,-12.05858zm-7.53661,12.05858l-5.42636,0l0,-12.05858l5.42636,0l0,12.05858zm1.20586,0l0,-12.05858l5.1249,0l0,12.05858l-5.1249,0zm-7.83808,0l-5.1249,0l0,-12.05858l5.1249,0l0,12.05858z" />
	</SVG>
);

export const mergeCells = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="m0,0l24,0l0,24l-24,0l0,-24z" fill="none" />
		<Polygon id="svg_2" points="21.5,18 2.5,18 2.5,20 21.5,20 21.5,18 " />
		<Path d="m19.5,10l0,4l-15,0l0,-4l15,0m1,-2l-17,0c-0.55,0 -1,0.45 -1,1l0,6c0,0.55 0.45,1 1,1l17,0c0.55,0 1,-0.45 1,-1l0,-6c0,-0.55 -0.45,-1 -1,-1l0,0z" />
		<Polygon id="svg_4" points="21.5,4 2.5,4 2.5,6 21.5,6 21.5,4 " />
	</SVG>
);
