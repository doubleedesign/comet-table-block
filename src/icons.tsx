/**
 * WordPress dependencies
 */
import { Path, SVG, Polygon } from '@wordpress/components';

export const blockIcon = (
	<SVG viewBox="0 0 24 24">
		<Polygon points="17.5 1 7.8 5.6 10.8 6 6.5 10 16.5 5.5 13.4 5 17.5 1" />
		<Path d="M19,11H5c-1.1,0-2,.9-2,2v8c0,1.1.9,2,2,2h14c1.1,0,2-.9,2-2v-8c0-1.1-.9-2-2-2ZM11.5,16.5h-7v-3.5c0-.3.2-.5.5-.5h6.5v4ZM13,12.5h6c.3,0,.5.2.5.5v3.5h-6.5v-4ZM4.5,21v-3h7v3.5h-6.5c-.3,0-.5-.2-.5-.5ZM19,21.5h-6v-3.5h6.5v3c0,.3-.2.5-.5.5Z" />
	</SVG>
);

export const alignTop = (
	<SVG viewBox="0 0 24 24">
		<Path d="M9 20h6V9H9v11zM4 4v1.5h16V4H4z" />
	</SVG>
);

export const alignMiddle = (
	<SVG viewBox="0 0 24 24">
		<Path d="M20 11h-5V4H9v7H4v1.5h5V20h6v-7.5h5z" />
	</SVG>
);

export const alignBottom = (
	<SVG viewBox="0 0 24 24">
		<Path d="M15 4H9v11h6V4zM4 18.5V20h16v-1.5H4z" />
	</SVG>
);

export const tableRowBefore = (
	<SVG viewBox="0 0 24 24">
		<Path d="M16,8h-3v-3s-2,0-2,0v3h-3s0,2,0,2h3v3s2,0,2,0v-3h3s0-2,0-2ZM4,2h16c1.1,0,2,.9,2,2v16c0,1.1-.9,2-2,2H4c-1.1,0-2-.9-2-2V4c0-1.1.9-2,2-2ZM20.5,14.5V4c0-.3-.2-.5-.5-.5H4c-.3,0-.5.2-.5.5v10.5s17,0,17,0ZM20.4,20v-4s-4.7,0-4.7,0v4.5s4.3,0,4.3,0c.3,0,.5-.2.4-.5ZM14.2,20.5v-4.5s-4.5,0-4.5,0v4.5s4.5,0,4.5,0ZM8.2,20.5v-4.5s-4.8,0-4.8,0v4c0,.3.2.5.5.5h4.2s.1,0,.1,0Z" />
	</SVG>
);

export const tableRowAfter = (
	<SVG viewBox="0 0 24 24">
		<Path d="M8,16h3v3s2,0,2,0v-3h3s0-2,0-2h-3v-3s-2,0-2,0v3h-3s0,2,0,2ZM20,22H4c-1.1,0-2-.9-2-2V4c0-1.1.9-2,2-2h16c1.1,0,2,.9,2,2v16c0,1.1-.9,2-2,2ZM3.5,9.5v10.5c0,.3.2.5.5.5h16c.3,0,.5-.2.5-.5v-10.5s-17,0-17,0ZM3.6,4v4s4.7,0,4.7,0V3.5s-4.3,0-4.3,0c-.3,0-.5.2-.4.5ZM9.8,3.5v4.5s4.5,0,4.5,0V3.5s-4.5,0-4.5,0ZM15.8,3.5v4.5s4.8,0,4.8,0v-4c0-.3-.2-.5-.5-.5h-4.2s-.1,0-.1,0ZZ" />
	</SVG>
);

export const tableColumnBefore = (
	<SVG viewBox="0 0 24 24">
		<Path d="M8,8v3h-3s0,2,0,2h3v3s2,0,2,0v-3h3s0-2,0-2h-3v-3s-2,0-2,0ZM2,20V4c0-1.1.9-2,2-2h16c1.1,0,2,.9,2,2v16c0,1.1-.9,2-2,2H4c-1.1,0-2-.9-2-2ZM14.5,3.5H4c-.3,0-.5.2-.5.5v16c0,.3.2.5.5.5h10.5s0-17,0-17ZM20,3.6h-4s0,4.7,0,4.7h4.5s0-4.3,0-4.3c0-.3-.2-.5-.5-.4ZM20.5,9.8h-4.5s0,4.5,0,4.5h4.5s0-4.5,0-4.5ZM20.5,15.8h-4.5s0,4.8,0,4.8h4c.3,0,.5-.2.5-.5v-4.2s0-.1,0-.1Z" />
	</SVG>
);

export const tableColumnAfter = (
	<SVG viewBox="0 0 24 24">
		<Path d="M16,16v-3h3s0-2,0-2h-3v-3s-2,0-2,0v3h-3s0,2,0,2h3v3s2,0,2,0ZM22,4v16c0,1.1-.9,2-2,2H4c-1.1,0-2-.9-2-2V4c0-1.1.9-2,2-2h16c1.1,0,2,.9,2,2ZM9.5,20.5h10.5c.3,0,.5-.2.5-.5V4c0-.3-.2-.5-.5-.5h-10.5s0,17,0,17ZM4,20.4h4s0-4.7,0-4.7H3.5s0,4.3,0,4.3c0,.3.2.5.5.4ZM3.5,14.2h4.5s0-4.5,0-4.5H3.5s0,4.5,0,4.5ZM3.5,8.2h4.5s0-4.8,0-4.8h-4c-.3,0-.5.2-.5.5v4.2s0,.1,0,.1Z" />
	</SVG>
);

export const tableRowDelete = (
	<SVG viewBox="0 0 24 24">
		<Path d="M12,14.8l-1.5,1.5h3s-1.5-1.5-1.5-1.5ZM12,9.2l-1.5-1.5h3s-1.5,1.5-1.5,1.5ZM20,2H4c-1.1,0-2,.9-2,2v16c0,1.1.9,2,2,2h16c1.1,0,2-.9,2-2V4c0-1.1-.9-2-2-2ZM20.5,7.8h-3s.7.7.7.7l-.8.7h3v5.5h-3s.7.7.7.7l-.8.7h3v3.8c0,.3-.2.5-.5.5H4c-.3,0-.5-.2-.5-.5v-3.8h3s-.7-.7-.7-.7l.8-.7h-3v-5.5h3s-.7-.7-.7-.7l.8-.7h-3v-3.8c0-.3.2-.5.5-.5h16c.3,0,.5.2.5.5v3.8ZM17,8.5l-3.5,3.5,3.5,3.5-1.5,1.5-3.5-3.5-3.5,3.5-1.5-1.5,3.5-3.5-3.5-3.5,1.5-1.5,3.5,3.5,3.5-3.5s1.5,1.5,1.5,1.5Z" />
	</SVG>
);

export const tableColumnDelete = (
	<SVG viewBox="0 0 24 24">
		<Path d="M9.2,12l-1.5-1.5v3l1.5-1.5ZM14.8,12l1.5-1.5v3l-1.5-1.5ZM22,20V4c0-1.1-.9-2-2-2H4c-1.1,0-2,.9-2,2v16c0,1.1.9,2,2,2h16c1.1,0,2-.9,2-2ZM16.2,20.5v-3l-.8.8-.8-.8v3h-5.5v-3l-.8.8-.8-.8v3h-3.8c-.3,0-.5-.2-.5-.5V4c0-.3.2-.5.5-.5h3.8v3l.8-.8.8.8v-3h5.5v3l.8-.8.8.8v-3h3.8c.3,0,.5.2.5.5v16c0,.3-.2.5-.5.5h-3.8ZM15.5,17l-3.5-3.5-3.5,3.5-1.5-1.5,3.5-3.5-3.5-3.5,1.5-1.5,3.5,3.5,3.5-3.5,1.5,1.5-3.5,3.5,3.5,3.5s-1.5,1.5-1.5,1.5Z" />
	</SVG>
);

export const tableMergeCell = (
	<SVG viewBox="0 0 24 24">
		<Path d="M2,20V4c0-1.1.9-2,2-2h16c1.1,0,2,.9,2,2v16c0,1.1-.9,2-2,2H4c-1.1,0-2-.9-2-2ZM8.2,3.5h-4.2c-.3,0-.5.2-.5.5v3.5h4.7V3.5ZM14.2,3.5h-4.5v4h4.5s0-4,0-4ZM20,3.5h-4.3v4h4.8v-3.5c0-.3-.2-.5-.5-.5ZM19,10.5H5v3h14v-3ZM8.2,16.5H3.5v3.5c0,.3.2.5.5.5h4.2v-4ZM14.2,16.5h-4.5v4h4.5s0-4,0-4ZM20.5,16.5h-4.8v4h4.3c.3,0,.5-.2.5-.5v-3.5Z" />
	</SVG>
);

export const tableSplitCell = (
	<SVG viewBox="0 0 24 24">
		<Path d="M2,20V4c0-1.1.9-2,2-2h16c1.1,0,2,.9,2,2v16c0,1.1-.9,2-2,2H4c-1.1,0-2-.9-2-2ZM4,3.5c-.3,0-.5.2-.5.5v3.5h17v-3.5c0-.3-.2-.5-.5-.5H4ZM15.2,13.5h3.8v-3h-3.8v3ZM10.2,10.5v3h3.5v-3h-3.5ZM8.7,10.5h-3.7v3h3.7v-3ZM20,20.5c.3,0,.5-.2.5-.5v-3.5H3.5v3.5c0,.3.2.5.5.5h16Z" />
	</SVG>
);

export const borderSeparate = (
	<SVG viewBox="0 0 24 24">
		<Path
			d="M6 5.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5H6a.5.5 0 01-.5-.5V6a.5.5 0 01.5-.5zM4 6a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2H6a2 2 0 01-2-2V6zm11-.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5h-3a.5.5 0 01-.5-.5V6a.5.5 0 01.5-.5zM13 6a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2h-3a2 2 0 01-2-2V6zm5 8.5h-3a.5.5 0 00-.5.5v3a.5.5 0 00.5.5h3a.5.5 0 00.5-.5v-3a.5.5 0 00-.5-.5zM15 13a2 2 0 00-2 2v3a2 2 0 002 2h3a2 2 0 002-2v-3a2 2 0 00-2-2h-3zm-9 1.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5H6a.5.5 0 01-.5-.5v-3a.5.5 0 01.5-.5zM4 15a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2H6a2 2 0 01-2-2v-3z"
			fillRule="evenodd"
			clipRule="evenodd"
		/>
	</SVG>
);

export const borderCollapse = (
	<SVG viewBox="0 0 24 24">
		<Path
			d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7.8 16.5H5c-.3 0-.5-.2-.5-.5v-6.2h6.8v6.7zm0-8.3H4.5V5c0-.3.2-.5.5-.5h6.2v6.7zm8.3 7.8c0 .3-.2.5-.5.5h-6.2v-6.8h6.8V19zm0-7.8h-6.8V4.5H19c.3 0 .5.2.5.5v6.2z"
			fillRule="evenodd"
			clipRule="evenodd"
		/>
	</SVG>
);

export const borderSolid = (
	<SVG viewBox="0 0 24 24">
		<Path d="M5 11.25h14v1.5H5z" />
	</SVG>
);

export const borderDotted = (
	<SVG viewBox="0 0 24 24">
		<Path
			fillRule="evenodd"
			d="M5.25 11.25h1.5v1.5h-1.5v-1.5zm3 0h1.5v1.5h-1.5v-1.5zm4.5 0h-1.5v1.5h1.5v-1.5zm1.5 0h1.5v1.5h-1.5v-1.5zm4.5 0h-1.5v1.5h1.5v-1.5z"
			clipRule="evenodd"
		/>
	</SVG>
);

export const borderDashed = (
	<SVG viewBox="0 0 24 24">
		<Path
			fillRule="evenodd"
			d="M5 11.25h3v1.5H5v-1.5zm5.5 0h3v1.5h-3v-1.5zm8.5 0h-3v1.5h3v-1.5z"
			clipRule="evenodd"
		/>
	</SVG>
);

export const borderDouble = (
	<SVG viewBox="0 0 24 24">
		<Path d="M5 9.25h14v1.5H5z" />
		<Path d="M5 13.25h14v1.5H5z" />
	</SVG>
);
