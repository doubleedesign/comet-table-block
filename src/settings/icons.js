/**
 * WordPress dependencies
 */
import { Path, SVG } from '@wordpress/components';

export const borderSeparate = (
	<SVG width="24" height="24">
		<Path
			d="M6 5.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5H6a.5.5 0 01-.5-.5V6a.5.5 0 01.5-.5zM4 6a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2H6a2 2 0 01-2-2V6zm11-.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5h-3a.5.5 0 01-.5-.5V6a.5.5 0 01.5-.5zM13 6a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2h-3a2 2 0 01-2-2V6zm5 8.5h-3a.5.5 0 00-.5.5v3a.5.5 0 00.5.5h3a.5.5 0 00.5-.5v-3a.5.5 0 00-.5-.5zM15 13a2 2 0 00-2 2v3a2 2 0 002 2h3a2 2 0 002-2v-3a2 2 0 00-2-2h-3zm-9 1.5h3a.5.5 0 01.5.5v3a.5.5 0 01-.5.5H6a.5.5 0 01-.5-.5v-3a.5.5 0 01.5-.5zM4 15a2 2 0 012-2h3a2 2 0 012 2v3a2 2 0 01-2 2H6a2 2 0 01-2-2v-3z"
			fillRule="evenodd"
			clipRule="evenodd"
		/>
	</SVG>
);

export const borderCollapse = (
	<SVG width="24" height="24">
		<Path
			d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7.8 16.5H5c-.3 0-.5-.2-.5-.5v-6.2h6.8v6.7zm0-8.3H4.5V5c0-.3.2-.5.5-.5h6.2v6.7zm8.3 7.8c0 .3-.2.5-.5.5h-6.2v-6.8h6.8V19zm0-7.8h-6.8V4.5H19c.3 0 .5.2.5.5v6.2z"
			fillRule="evenodd"
			clipRule="evenodd"
		/>
	</SVG>
);
