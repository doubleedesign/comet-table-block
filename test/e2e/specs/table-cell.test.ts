/**
 * WordPress dependencies
 */
import {
	getEditedPostContent,
	createNewPost,
	clickButton,
	clickBlockToolbarButton,
	pressKeyWithModifier,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	flexibleTableCellSelector,
	createNewFlexibleTableBlock,
	clickButtonWithAriaLabel,
	clickButtonWithText,
	clickToggleControlWithText,
	openSidebar,
	getWpVersion,
} from '../helper';
import { link } from '@wordpress/icons';

/** @type {import('puppeteer').Page} */
const page = global.page;

describe( 'Flexible table cell', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'allows cell movement with tab key.', async () => {
		const wpVersion = await getWpVersion();
		await createNewFlexibleTableBlock();
		await openSidebar();
		await clickButton( 'Global setting' );

		// Restore settings.
		await clickButton( 'Restore default settings' );
		await clickButtonWithText(
			'//div[contains(@class,"ftb-global-setting-modal__confirm-popover")]',
			'Restore'
		);
		await page.waitForSelector( '.ftb-global-setting-modal__notice' );
		await clickButtonWithAriaLabel( '.ftb-global-setting-modal__notice', 'Dismiss this notice' );

		await clickButton( 'Editor options' );
		await clickToggleControlWithText( 'Use the tab key to move cells' );
		await clickButtonWithText(
			'//div[@class="ftb-global-setting-modal__buttons"]',
			'Save setting'
		);
		await page.waitForSelector( '.ftb-global-setting-modal__notice' );
		const modalCloseLabel = [ '6-2', '6-3', '6-4' ].includes( wpVersion )
			? 'Close'
			: 'Close dialog';
		await clickButtonWithAriaLabel( '.ftb-global-setting-modal', modalCloseLabel );
		const cells = await page.$$( flexibleTableCellSelector );
		await cells[ 0 ].click();
		await page.keyboard.type( 'Cell 1' );
		await page.keyboard.down( 'Tab' );
		await page.keyboard.up( 'Tab' );
		await page.keyboard.down( 'Tab' );
		await page.keyboard.up( 'Tab' );
		await page.keyboard.down( 'Shift' );
		await page.keyboard.down( 'Tab' );
		await page.keyboard.up( 'Tab' );
		await page.keyboard.up( 'Shift' );
		await page.keyboard.type( 'Cell 2' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'allows keyboard operation within the link popover', async () => {
		await createNewFlexibleTableBlock();
		const cells = await page.$$( flexibleTableCellSelector );
		await cells[ 0 ].click();
		await page.keyboard.type( 'Link' );
		await pressKeyWithModifier( 'primary', 'a' );
		await clickBlockToolbarButton( 'Link' );

		// Create a link.
		await page.keyboard.down( 'Shift' );
		await page.keyboard.type( '#anchor' );
		await page.keyboard.up( 'Shift' );
		await page.keyboard.press( 'Enter' );

		expect( await getEditedPostContent() ).toMatchSnapshot();

		// Edit the link.
		await page.keyboard.press( 'ArrowLeft' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( '-updated' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( '#anchor-updated' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'Space' );
		await clickButtonWithText( '//div[contains(@class,"components-popover")]', 'Save' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
