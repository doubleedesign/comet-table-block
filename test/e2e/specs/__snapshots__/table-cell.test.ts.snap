// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexible table cell allows cell movement with tab key. 1`] = `
"<!-- wp:flexible-table-block/table -->
<figure class="wp-block-flexible-table-block-table"><table class="has-fixed-layout"><tbody><tr><td>Cell 1</td><td>Cell 2</td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Flexible table cell allows keyboard operation within the link popover 1`] = `
"<!-- wp:flexible-table-block/table -->
<figure class="wp-block-flexible-table-block-table"><table class="has-fixed-layout"><tbody><tr><td><a href="#anchor">Link</a></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Flexible table cell allows keyboard operation within the link popover 2`] = `
"<!-- wp:flexible-table-block/table -->
<figure class="wp-block-flexible-table-block-table"><table class="has-fixed-layout"><tbody><tr><td><a href="#anchor-updated" target="_blank" rel="noreferrer noopener">Link-updated</a></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;
